employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/otp/OtpService.java

```diff
@@ -0,0 +1,16 @@
+package org.optaweb.employeerostering.service.otp;
+
+import org.optaweb.employeerostering.domain.otp.OneTimePassword;
+import org.optaweb.employeerostering.exception.OtpCreationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OtpService {
+    @Autowired
+    OtpRepository otpRepository;
+
+    public OneTimePassword createOtp(String email) throws OtpCreationException {
+        return otpRepository.save(new OneTimePassword(email));
```

Comment:
Did you mean to throw an `OtpCreationException` here on failure? Thinking about it.... what about using Spring's `AuthenticationServiceException` instead?